test/utils.sh

```diff
@@ -227,3 +227,88 @@ get_image_manifests() {
   echo "${image_manifests}"
 
 }
+
+# This function will be used by tekton tasks in build-definitions
+# It returns a list of unreleased bundles in the FBC fragment by comparing it with
+# the corresponding production index image
+get_unreleased_bundle() {
+  # FBC fragment containing the unreleased bundle
+  local FBC_FRAGMENT=""
+
+  get_unreleased_bundle_usage()
+  {
+    echo "
+  get_unreleased_bundle  -i FBC_FRAGMENT
+  " >&2
+    exit 2
+  }
+
+  local OPTIND opt
+  while getopts "i:" opt; do
+      case "${opt}" in
+          i)
+              FBC_FRAGMENT="${OPTARG}" ;;
+          *)
+              get_unreleased_bundle_usage ;;
+      esac
+  done
+  shift $((OPTIND-1))
+
+  if [ -z "$FBC_FRAGMENT" ]; then
+    echo "Missing parameter FBC_FRAGMENT" >&2
+    exit 2
+  fi
+
+  local INDEX_IMAGE="registry.redhat.io/redhat/redhat-operator-index"
+
+  #Get target ocp version from the fragment
+  if ! ocp_version=$(skopeo inspect --no-tags --raw docker://"$FBC_FRAGMENT"); then
+    echo "The raw image inspect command failed" >&2
+    exit 1
+  fi
+  ocp_version=$(echo "$ocp_version" |  jq -r '.annotations."org.opencontainers.image.base.name"' | cut -d: -f2)
+
+  # Decide which OPM version to use for rendering the image
+  local opm_version
+  if [[ "$(printf '%s\n' "$ocp_version" "v4.17" | sort -V | head -n1)" == "v4.17" ]]; then
+    opm_version="opm-v1.40.0"
+  else
+    opm_version="opm"
+  fi
+
+
+  # Run opm render on the FBC fragment to extract package name and unique bundles
+  local render_out_fbc unique_bundles_fbc package_name
+  if ! render_out_fbc=$($opm_version render "$FBC_FRAGMENT"); then
```

Comment:
We also do this in the FBC tasks. Would it make sense to create some tool abstraction again? 
https://github.com/konflux-ci/build-definitions/blob/main/task/fbc-related-image-check/0.1/fbc-related-image-check.yaml#L37

Reply:
I don't think we should create abstraction just for rendering. It's already abstracted by opm. However, I am open to abstracting package name extraction. I'll make that change. Please let me know if you feel strongly about abstracting `opm render`.

Reply:
Done